### DISCUSSION TITLE
Internal GPS

### DISCUSSION THREAD

**Discussion by stevempotter** [EdgeTX/edgetx#3436] [Q&A]
I am trying to get the internal GPS I installed in my Radiomaster TX16S (with EdgeTX 2.8.2) to display GPS coordinates of the radio in a widget. Does anyone know how to do that?
Also, I see only NMEA and DMS options for the formatting of the coordinates. I would prefer just decimal degrees, such as
54.1234567, -6.1234567
as these are much easier to type into Google Maps if needed. Can you add that to the next version of EdgeTX?

I can see a little satellite icon in the top bar that shows me how many satellites it is seeing, and can see it is working in the Statistics:Debug window.

**Comment by pfeerick** [comment]
Maybe have a look at some of the Lua GPS widgets - https://github.com/moschotto/OpenTX_GPS_Telemetry and https://github.com/kristjanbjarni/opentx-widgets

**Reply by rotorman** [reply]
Included on stock EdgeTX SD card is a radio GPS test widget that you can tweak for your purpose: https://github.com/EdgeTX/edgetx-sdcard/blob/master/sdcard/c480x272/WIDGETS/TxGPStest/main.lua

**Comment by smcStevesmc** [comment]
Therer are 4 pads for rx/tx/3v/grnd you can solder on inthe board. Remove 4 screws from back case and two from next to the antenna, not too hard.

HOWEVER, I installed a HGLRC M100 but it isnt working.. I trried the same 2 units on my drone and they worked fine, but my TX16S just isn't recieving data....Itried swapping rx and tx too..any ideas?

**Reply by rotorman** [reply]
Can you first try to hook up your HGLRC M100 GPS via FTDI cable to your radio and verify with a terminal program, such as [hTerm](https://www.der-hammer.info/pages/terminal.html), that it really is outputting NMEA at 9600 baud 8N1.
The product page lists 115200 baud and UBLOX protocol as default for this GPS: https://www.hglrc.com/products/m100-5883-gps and I remember reading that some uBlox M10 based GPS do not allow settings to be saved. You could test also with [u-center 2](https://www.u-blox.com/en/product/u-center) software.

And EdgeTX would surely be happy to receive a PR with uBlox protocol support. For hints, how this could be done could be taken e.g. from: https://github.com/betaflight/betaflight/blob/master/src/main/io/gps.c

**Reply by pfeerick** [reply]
FYI, this is a cross post as was also asked in #4562

**Reply by DHaacke** [reply]
Adding to the above advice and FWIW, and hoping it saves someone else some time, I installed a UBLOX SAM-M10Q (no Magnetometer) in my new TX16S Max just the other day. Here are some findings. 

It was pure luck it came with a long JST SH1.0 cable with connectors at both end as it plugged right in to the UART5 4 pin socket (which EdgeTx refers to as AUX1). The bad news was the default baud rate was 9600. I already had u-center2 installed on my Mac (running Parallels Desktop). I had to wire up another JST SH1.0 plug so I could attach it to a serial TTL converter to talk to u-center2.  It connected easy enough, but took some poking around until I went in the device configuration and was able to change the baud rate to 115200. The KEY is after changing the baud rate, you MUST save the configuration (there's a small text button). If you make the change and don't save, the baud rate will be lost after power down. Got it working great now.  Good luck!
